src/realm.h

```diff
@@ -2939,6 +2939,50 @@ RLM_API void realm_clear_cached_apps(void) RLM_API_NOEXCEPT;
 RLM_API const char* realm_app_get_app_id(const realm_app_t*) RLM_API_NOEXCEPT;
 RLM_API realm_user_t* realm_app_get_current_user(const realm_app_t*) RLM_API_NOEXCEPT;
 
+/**
+ * Update the URL used to communicate with the Realm server. This function will update the location
+ * information used for http and websocket requests to the server. Once this operation has completed,
+ * the new base_url value returned by realm_app_get_base_url() will match the base_url value provided
+ * to this function. Any App requests performed while the base URl update is currently in progress
+ * will continue to use the original base URL value.
+ *
+ * @param app ptr to realm_app
+ * @param base_url The new base URL value to set as the Realm server URL.
+ * @param callback invoked once operation has completed
+ * @return True if no error has been recorded, False otherwise
+ */
+RLM_API bool realm_app_update_base_url(realm_app_t* app, realm_string_t base_url,
+                                       realm_app_void_completion_func_t callback, realm_userdata_t userdata,
+                                       realm_free_userdata_func_t userdata_free);
+
+/**
+ * Update the URL used to communicate with the Realm server to the default value (same as if a base_url
+ * value was not provided in AppConfig). This function will update the location information used for http
+ * and websocket requests to the server. Once this operation has completed, the new base_url value
+ * returned by realm_app_get_base_url() will match the base_url value provided to this function. Any App
+ * requests performed while the base URl update is currently in progress will continue to use the
+ * original base URL value.
+ *
+ * @param app ptr to realm_app
+ * @param callback invoked once operation has completed
+ * @return True if no error has been recorded, False otherwise
+ */
+RLM_API bool realm_app_set_default_base_url(realm_app_t* app, realm_app_void_completion_func_t callback,
```

Comment:
why is this needed? is it to overwrite any base_url provided in AppConfig?

Reply:
It is to be able to reset the base URL value to the default base url value, in case the default base url is not known to the SDK dev. e.g.

* Start App with default base url
* Change base url using `update_base_url()` to connect to an Edge Server.
* Revert base url to default to connect to the cloud again by calling this function or `App::update_base_url(std::nullopt, ...)` 

I don't think the default base url value is exposed as a const to the SDK developer anywhere...

Reply:
got it. It seems the user can just use `realm_app_update_base_url` with nullopt. We could add a note to realm_app_update_base_url that std::nullopt is used to specify the default base_url. Your call.

Reply:
I updated the C_API function to take a `const char*` and the function will use the default base URL if the value is null or an empty string. I also added some tests to verify this operation.